src/components/ContentSidebar/MetadataSidebar.js

```diff
@@ -110,9 +154,23 @@ class MetadataSidebar extends React.PureComponent<Props, State> {
     onAdd = (template: MetadataEditorTemplate) => {
         const { api, file }: Props = this.props;
         this.setState({ isLoading: true });
-        api.getMetadataAPI(false).createMetadata(file, template, this.getMetadataEditors, this.errorCallback);
+        api.getMetadataAPI(false).createMetadata(file, template, this.onAddSuccessHandler, this.errorCallback);
     };
 
+    /**
+     * Instance save success handler
+     *
+     * @param {Object} editor - updated editor
+     * @param {Array} editors - array of editors
+     * @param {number} index - editor to remove index
+     * @return {void}
+     */
+    onSaveSuccessHandler(editor: MetadataEditor, editors: Array<MetadataEditor>, index: number): void {
```

Comment:
Minor: Passing the index around feels a little clunky. Can you pass the editor itself and use `indexOf` inside the functions to achieve the same result?